core/src/main/java/org/apache/calcite/sql2rel/SqlToRelConverter.java

```diff
@@ -3445,6 +3446,25 @@ private Pair<RexNode, RelNode> convertOnCondition(
         }
 
         old_right_rel_count = cur_right_rel_count;
+
+        RexNode oldNodeExpr = node.expr;
+        if (oldNodeExpr instanceof RexRangeRef) {
```

Comment:
How stable would you say this is? Are you worried this is too hacky?

Reply:
It is stable specifically for scalar sub queries, they ALWAYS result in a RexRangeRef with an offset of 0, it's commented later in the code that handles the substitution. This isn't necessarily stable across all possible sub queries (though none of them worked before either), we'll need an actual plan to get 100% coverage.

Reply:
Okay sounds good. Can you file an issue for working on these changes + a confluence docs adding some additional context?

Reply:
Can you add the issue and/or confluence doc before we merge this.